packages/core-types/src/SmartAccountTypes.ts

```diff
@@ -58,6 +58,7 @@ export type SmartAccountState = {
   owner: string // multichain (EVM)
   isDeployed: boolean // chain specific
   entryPointAddress: string // chain specific?
+  implementationAddress: string
```

Comment:
will the state be reset if implementation is upgraded? 
same for isDeployed. it should not be just set once